rootsanalyzer/rootsanalyzer.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package rootsanalyzer reports on changes in the set of root certificates
+// returned by a CT Log's get-roots endpoint.
+package rootsanalyzer
+
+import (
+	"bytes"
+	"context"
+	"net/url"
+	"path"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/golang/glog"
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/monologue/ctlog"
+	"github.com/google/monologue/incident"
+	"github.com/google/monologue/storage"
+)
+
+var (
+	incidentTemplate = template.Must(template.New("roots_incident").Parse(`The root certificates accepted by {{ .Log.Name }} ({{ .Log.URL }}) have changed.
+{{ if gt (len .AddedCerts) 0 }}
+Certificates added ({{ len .AddedCerts }}):
+{{ range .AddedCerts }}{{ .Subject }}
+{{ end }}{{ end }}{{ if gt (len .RemovedCerts) 0 }}
+Certificates removed ({{ len .RemovedCerts }}):
+{{ range .RemovedCerts }}{{ .Subject }}
+{{ end }}{{ end }}`))
```

Comment:
If only the subject of the certs will be going in to the incident reports, it's almost definitely worth storing the complete cert diff data along side the RootSetID or something (as mentioned in another comment), just in case there end up being roots with the same subject field, but different keys, or something?  It'd be good to have the full diff info somewhere I think?

Reply:
I could include the full diff in here; I can imagine that being useful.

Reply:
For now, I've just included the SHA256 hash of the cert, which makes it clear if something about the cert has changed (if it's in both the added and removed sections). That hash can be thrown into https://crt.sh to get the full cert details. I think the full diff could be deferred for a future PR.